### PR TITLE
HTML Styling improvements

### PR DESCRIPTION
**Pull request by michaelnt**
These are the styling improvements from rickeywang plus:

- a fix for the missing headings, when an item doesn't have a header attribute
- a fix to the heading levels (but also don't limit to only two level)
- fix the tests (hopefully)

### CODE REVIEW

**Comment by rickeywang** [comment]
Hi, thanks for fixing my PR. I got quite busy lately and was just about to start fixing all the problems. 

I think `doorstop.css` isn't used anywhere? Especially now that you got rid of the `yield from _lines_css()` in `doorstop/core/publisher.py`. 

To address https://github.com/jacebrowning/doorstop/pull/353#issuecomment-508889445 , I think we can put a `max-width: 960px` for the body of the Index page. Perhaps it makes sense to have a separate CSS file for the index page?

File: doorstop/core/publisher.py [hunk 1]
```diff
@@ -154,9 +154,8 @@ def _lines_index(filenames, charset='UTF-8', tree=None):
         '<meta http-equiv="content-type" content="text/html; '
         'charset={charset}">'.format(charset=charset)
     )
-    yield '<style type="text/css">'
-    yield from _lines_css()
-    yield '</style>'
+    yield '<link rel="stylesheet" href="assets/doorstop/bootstrap.min.css" />'
+    yield '<link rel="stylesheet" href="assets/doorstop/general.css" />'
```
Comment (rickeywang) [inline doorstop/core/publisher.py]:
```suggestion
    yield '<link rel="stylesheet" href="assets/doorstop/general.css" />'
    yield '<title>Doorstop Index</title>'
```

Right now the Index page has no Title. Give it a title.

File: doorstop/core/publisher.py [hunk 2]
```diff
@@ -189,14 +188,14 @@ def _lines_index(filenames, charset='UTF-8', tree=None):
         # table
         yield '<h3>Item Traceability:</h3>'
```
Comment (rickeywang) [inline doorstop/core/publisher.py]:
```suggestion
        yield '<h3>Item Traceability</h3>'
```

**Comment by rickeywang** [comment]
Also I think we can remove the colon `:` after each headings "Tree Structure", "Published Documents" and "Item Traceability" to make the index cleaner.

**Comment by nhartung** [comment]
Does anyone know the status of this PR? If we still think these updates are desirable, I can fix the merge conflicts and review the changes. I'm interested in diving into the publisher code and this will give me a good opportunity to do that.

**Comment by rickeywang** [comment]
> 
> 
> Does anyone know the status of this PR? If we still think these updates are desirable, I can fix the merge conflicts and review the changes. I'm interested in diving into the publisher code and this will give me a good opportunity to do that.

I would like to see this merged too but it's on @michaelnt 's branch so I didn't want to be touching other people's work. Maybe you can branch off of michaelnt:styling to make the necessary changes? Or better yet make the branch here in the doorstop-dev/doorstop repo

**Comment by nhartung** [comment]
Thanks for the info @rickeywang. I'll go ahead and create a branch on my fork and start working on this.

**Comment by jacebrowning** [comment]
I'm going to close this due to inactivity. Feel free to open a new PR!

**Pull request closed without merging**